posts/cassandra_models_benchmarking.md

```diff
@@ -0,0 +1,150 @@
+    * Collection of good practices given by [Patrick McFadin](https://twitter.com/patrickmcfadin) at last Cassandra Summit EU (2014)
+* [16 Notes on Cassandra Summit (II)](http://mrcalonso.com/16-notes-cassandra-summit-europe-2014-part-ii/)
+    * Collection of some more good practices and brand new features to consider. Again from Cassandra Summit EU 2014
+* [Cassandra Day London 2015](http://mrcalonso.com/cassandra-day-london-2015-2/)
+    * Collection of some use cases and a very interesting anti-patterns list by [Christopher Batey](https://twitter.com/chbatey)
+* [Getting Started with Cassandra Time Series data modelling](http://patrickmcfadin.com/2014/02/05/getting-started-with-time-series-data-modeling/)
+    * Three different patterns you can follow when using Cassandra to store time series
+
+Once you've all the ideas fresh in your mind is time to start thinking about your data model.
+
+### 2. Design your data model and alternatives
+
+Following all the ideas and principles learnt in the previous step, design your data model and
+try to think of alternatives. These alternatives usually come as minor adjustments that can
+be applied to the model but just by following the best practices you can't decide whether one
+or the other is a better choice.
+
+Here I'll provide two examples we've recently had:
+
+1. Having a bucketed time series with a maximum of 86.400 rows per partition, how is it better to read an entire partition?
+    a) By reading the whole partition at once
+    b) By reading the whole partition in chunks (2 halves, 4 quarters, ...)
+2. Having a model that contains the information of a discretised distribution on each record, how is it better to save the bins?
+    a) By using a *List* element that will contain the 100 bins
+    b) By having 100 columns, one for each bin
+
+The resulting models will meet all the good practices and avoid all the anti-patterns regardless
+of the final decission so, how do you decide which way to go?
+
+### 3. Benchmark your alternatives
+
+For this purpose I've created a script (Ruby in this case) that:
+
+1. Creates the table purposed by the model under test
+2. Generates **PRODUCTION DATA** and saves it (memory or disk, depending on the size)
+3. Benchmarks the applicable access patterns, in our case:
+    3.1. Insertion of all the data generated in step 2
+    3.2. Reading of all the data
+    3.3. Updating the data
+
+It is important that the access patterns are exactly the same way they'll be in
+production, otherwise the result of the benchmark is completely useless.
+
+This script should be adapted and ran for every single alternative.
+
+Here you can see the scripts used for both alternatives proposed for the example No. 2 described above:
+
+* [Benchmark the model with the list](https://gist.github.com/calonso/6dba841659000fdd7959)
+* [Benchmark the model with columns](https://gist.github.com/calonso/f14a0d9673110d6f9578)
```

Comment:
Could you obfuscate the code in these examples please? Replace anything which might refer to internals, e.g. `rc` and `sub_id`